src/sagas/demo.test.js

```diff
@@ -0,0 +1,76 @@
+import { put, select } from "redux-saga/effects";
+import { getDemoDocument } from "../reducers/demo";
+import { verifyDemoDocument } from "./demo";
+import {
+  whenDocumentValidAndIssuedByDns,
+  whenDocumentHashInvalidAndNotIssued,
+} from "../test/fixture/verifier-responses";
+import { history } from "../history";
+
+jest.mock("../services/verify", () => ({ verifyDocument: () => {} }));
+
+describe("verifyDemoDocument", () => {
```

Comment:
can we try to write sagas tests as run to completion kind instead of the step by step way?
https://dev.to/phil/the-best-way-to-test-redux-sagas-4hib
then we test for what actions have been dispatched or what services we have called.

Reply:
ok updated